sonar-plugin-api/src/main/java/org/sonar/api/utils/text/JsonWriter.java

```diff
@@ -381,15 +381,12 @@ public JsonWriter prop(String name, double value) {
   }
 
   /**
-   * @throws org.sonar.api.utils.text.WriterException on any failure
+   * Do nothing. The caller giving the Writer should manage the resource.
+   * The com.google.gson.stream.JsonWriter resource is garbage collected.
    */
   @Override
   public void close() {
-    try {
-      stream.close();
-    } catch (Exception e) {
-      throw rethrow(e);
-    }
+    // do nothing
```

Comment:
I've just changed my mind on this one. The JsonWriter behavior should be the same as java.io.Writer which means that underlying resources are closed. Moreover, we have the risk of having non flushed data lost